File: PHP/242. Valid Anagram.php
<?php
// 1. 242. Valid Anagram
// Level: Easy
// https://leetcode.com/problems/valid-anagram/

class Solution
{
  function isAnagram($s, $t)
  {
    $map = [];

    for ($i = 0; $i < strlen($s); $i++) {
      if (!array_key_exists($s[$i], $map))
        $map[$s[$i]] = 1;
      else
        $map[$s[$i]]++;
    }

    for ($i = 0; $i < strlen($t); $i++) {
      if (!array_key_exists($t[$i], $map))
        return false;
      else
        $map[$t[$i]]--;
    }

    foreach ($map as $key => $value) {
      if ($value != 0)
        return false;
    }
    return true;
  }
}

File: CSharp/242. Valid Anagram.cs
// 1. 242. Valid Anagram
// Level: Easy
// https://leetcode.com/problems/valid-anagram/


public class Solution
{
  public bool IsAnagram(string s, string t)
  {
    var map = new Dictionary<char, int>();

    for (var i = 0; i < s.Length; i++)
    {
      if (!map.ContainsKey(s[i]))
        map[s[i]] = 1;
      else
        map[s[i]]++;
    }

    for (var i = 0; i < t.Length; i++)
    {
      if (!map.ContainsKey(t[i]))
        return false;
      else
        map[t[i]]--;
    }

    foreach (var c in map)
    {
      if (c.Value != 0)
        return false;
    }
    return true;
  }
}